zapisy/apps/enrollment/records/models/opening_times.py

```diff
@@ -55,6 +56,34 @@ def is_after_t0(cls, student: Student, semester: Semester, time: datetime) -> bo
             return False
         return True
 
+    @staticmethod
+    def prevent_nighttime_records(semester: Semester, record_time: datetime) -> datetime:
+        """Prevention of night-time records.
+
+        Moves the opening time of records to daytime without rearranging
+        the order of ranked records.
+        """
+        how_many_12_hours_peroids_from_start = \
+            int((semester.records_opening - record_time) / timedelta(hours=12))
+
+        return record_time - (timedelta(hours=12) * how_many_12_hours_peroids_from_start)
```

Comment:
```suggestion
        how_many_12_hour_periods_from_start = \
            int((semester.records_opening - record_time) / timedelta(hours=12))

        return record_time - (timedelta(hours=12) * how_many_12_hour_periods_from_start)
```
Czy _half-day_ byłoby czytelnym zamiennikiem dla _12-hour period_?

Reply:
Reczywiście, to chyba lepsza nazwa.